src/algebra/lie/basic.lean

```diff
@@ -371,6 +447,55 @@ def inverse (f : M →ₗ⁅R,L⁆ N) (g : N → M)
               ... = ⁅x, g n⁆ : (h₁ _),
   ..linear_map.inverse f.to_linear_map g h₁ h₂ }
 
+instance : has_add (M →ₗ⁅R,L⁆ N) :=
+{ add := λ f g, { map_lie' := by simp, ..((f : M →ₗ[R] N) + (g : M →ₗ[R] N)) }, }
+
+instance : has_sub (M →ₗ⁅R,L⁆ N) :=
+{ sub := λ f g, { map_lie' := by simp, ..((f : M →ₗ[R] N) - (g : M →ₗ[R] N)) }, }
+
+instance : has_neg (M →ₗ⁅R,L⁆ N) :=
+{ neg := λ f, { map_lie' := by simp, ..(-(f : (M →ₗ[R] N))) }, }
+
+@[norm_cast, simp] lemma coe_add (f g : M →ₗ⁅R,L⁆ N) : ⇑(f + g) = f + g := rfl
+
+lemma add_apply (f g : M →ₗ⁅R,L⁆ N) (m : M) : (f + g) m = f m + g m := rfl
+
+@[norm_cast, simp] lemma coe_sub (f g : M →ₗ⁅R,L⁆ N) : ⇑(f - g) = f - g := rfl
+
+lemma sub_apply (f g : M →ₗ⁅R,L⁆ N) (m : M) : (f - g) m = f m - g m := rfl
+
+@[norm_cast, simp] lemma coe_neg (f : M →ₗ⁅R,L⁆ N) : ⇑(-f) = -f := rfl
+
+lemma neg_apply (f : M →ₗ⁅R,L⁆ N) (m : M) : (-f) m = -(f m) := rfl
+
+instance : add_comm_group (M →ₗ⁅R,L⁆ N) :=
+{ zero           := 0,
+  add            := (+),
+  neg            := λ f, -f,
+  sub            := λ f g, f - g,
+  add_assoc      := λ f g h, by { ext, simp only [add_assoc, add_apply], },
+  zero_add       := λ f, by { ext, simp only [add_apply, zero_add, zero_apply], },
+  add_zero       := λ f, by { ext, simp only [add_zero, add_apply, zero_apply], },
+  sub_eq_add_neg := λ f g, by { ext, simp only [sub_eq_add_neg, sub_apply, add_apply, neg_apply], },
+  add_left_neg   := λ f, by { ext, simp only [add_apply, add_left_neg, neg_apply, zero_apply], },
+  add_comm       := λ f g, by { ext, simp only [add_comm, add_apply], }, }
```

Comment:
Regarding @jcommelin's remark,
```suggestion
  sub_eq_add_neg := λ f g, by { ext, simp only [sub_eq_add_neg, sub_apply, add_apply, neg_apply], },
  add_left_neg   := λ f, by { ext, simp only [add_apply, add_left_neg, neg_apply, zero_apply], },
  .. lie_module_hom.add_comm_monoid}
```

Reply:
I could be missing the obvious but I don't see the connection to Johan's point about the diamonds here.

Are you suggesting I should define `lie_module_hom.add_comm_monoid` before I define `lie_module_hom.add_comm_group`? What is the benefit of this?

Reply:
No, I misread - ignore that comment!